ArduCopter/Parameters.cpp

```diff
@@ -1143,6 +1143,13 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     // @User: Standard
     AP_SUBGROUPINFO(command_model_pilot, "PILOT_Y_", 56, ParametersG2, AC_CommandModel),
 
+    // @Param: TKOFF_THRTIME
+    // @DisplayName: Ramp time of throttle during take-off
+    // @Description: This is the time in seconds to ramp the throttle from minimum to maximum during takeoff.
+    // @Range: 0.25 5.0
+    // @User: Standard
+    AP_GROUPINFO("TKOFF_THRTIME", 57, ParametersG2, takeoff_throttle_ramp_time, 2.0),
```

Comment:
i think the interaction with hover throttle in terms of the actual time to get off the ground is problematic from an end user point of view

Reply:
The user currently doesn't know how long it takes to get off the ground so that value is not relevant to the user. 
The actual time to takeoff currently changes each time during a manual takeoff based on the vertical velocity commanded by the pilot. 
In all modes the takeoff time depends on the PSC Z tuning and the vertical climb rate setpoint and weight. 
We also don't know what the actual hover throttle is due to change in payload, fuel load, or even inaccurate settings of hover throttle with learn turned off. (I set hover throttle to be the hover throttle at maximum payload with learn turned off as this is what I think the best balance given the various issues)

The goal here is to provide a consent takeoff behaviour that is tolerant to all the factors above and easy to understand and set for the user. The throttle rate of change is the single variable that we can use to define this independent of all other factors.

So we have two options, takeoff slew time or take off slew rate. 

I chose slew time because I could describe this to a user as "This is how long it takes to move the throttle from minimum to maximum while it is waiting for the aircraft to leave the ground. It is also the maximum time before the aircraft will give up trying to detect take-off and assume it is flying".

In reality most people will simply increase or decrease this number by a factor of 2 if they want a faster or slower take-off.

As a general rule I move from time constant to rates based on the nominal magnitude. So for this case we are on the edge there. We are talking throttle rates of 0.25 to 4 and ramp times or time constants or 0.25 to 4.